File: samples/sample.cpp
#include <vector>

int factorial(int n) {
    int result = 1;
    int i = 1;
    while (i <= n) {
        result *= i;
        i++;
    }
    return result;
}

int fibonacci(int n) {
    if (n <= 1) {
        return n;
    }
    int a = 0;
    int b = 1;
    for (int i = 2; i <= n; i++) {
        int tmp = a + b;
        a = b;
        b = tmp;
    }
    return b;
}

int classify(int value) {
    int label = 0;
    switch (value % 3) {
        case 0:
            label = 1;
            break;
        case 1:
            label = 2;
            break;
        default:
            label = 3;
            break;
    }
    return label;
}

int find_max(const std::vector<int>& values) {
    int max_val = values[0];
    for (int v : values) {
        if (v > max_val) {
            max_val = v;
        }
    }
    return max_val;
}

int count_positive(const std::vector<int>& values) {
    int count = 0;
    int i = 0;
    do {
        if (values[i] > 0) {
            count++;
        }
        i++;
    } while (i < (int)values.size());
    return count;
}

int main() {
    int r = factorial(5);
    int f = fibonacci(10);
    std::vector<int> nums = {3, 1, 4, 1, 5, 9, 2, 6};
    int biggest = find_max(nums);
    int pos = count_positive(nums);
    int label = classify(7);
    return 0;
}

File: samples/sample.py
def factorial(n):
    result = 1
    i = 1
    while i <= n:
        result *= i
        i += 1
    return result


def fibonacci(n):
    if n <= 1:
        return n
    a = 0
    b = 1
    for _ in range(2, n + 1):
        a, b = b, a + b
    return b


def classify(value):
    match value % 3:
        case 0:
            label = "fizz"
        case 1:
            label = "one"
        case _:
            label = "other"
    return label


def find_max(values):
    max_val = values[0]
    for v in values:
        if v > max_val:
            max_val = v
    return max_val


def count_positive(values):
    count = 0
    i = 0
    while i < len(values):
        if values[i] > 0:
            count += 1
        i += 1
    return count


result = factorial(5)
fib = fibonacci(10)
nums = [3, 1, 4, 1, 5, 9, 2, 6]
biggest = find_max(nums)
pos = count_positive(nums)
label = classify(7)
